Feat(core-component): Add attachment card component and styles

saadman30 opened the pull request:
Component: Attachement card
Clickup [Link](https://app.clickup.com/t/86ev9wr1q)

saadman30 committed:
Add attachment card component and styles

Introduced a new attachment card component with SCSS styles for displaying downloadable resources. The component includes metadata and a clear download state, enhancing the user experience for file management. Updated the index file to forward the new component and included it in the playground for demonstration.

saadman30 committed:
Refactor attachment card styles

Removed unnecessary margin from the title and cleaned up the badge styles in the attachment card component SCSS. This enhances the visual consistency and responsiveness of the component during hover interactions.

saadman30 committed:
Update playground template to conditionally include attachment card component

Removed the attachment card component from the initial section and added it back in a later section of the playground template. This change improves the organization of components and ensures the attachment card is displayed correctly in the demo.

b-l-i-n-d commented on assets/core/scss/components/_attachment-card.scss:
```diff
@@ -0,0 +1,67 @@
+    @include tutor-typography('small', 'medium', 'primary');
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
```
Use the mixing
```suggestion
    @include tutor-text-truncate;
```

saadman30 committed:
Merge branch '4.0.0-dev' into feat/core-attachment-card-component

saadman30 committed:
refactor: Enhance attachment card styles and structure

- Updated the SCSS for the attachment card to ensure full width and improved text truncation.
- Modified the HTML structure to include a base `tutor-card` class for consistency across components.
- Adjusted padding in demo component for better spacing.
